core/src/main/java/org/lflang/generator/c/CParameterGenerator.java

```diff
@@ -24,10 +27,36 @@ public static String getInitializer(ParameterInstance p) {
     if (p.getName().equals("bank_index") && p.getOverride() == null) {
       return CUtil.bankIndex(p.getParent());
     }
+    String initializer = null;
 
     CTypes ctypes = CTypes.generateParametersIn(p.getParent().getParent());
     Initializer values = p.getActualValue();
-    return ctypes.getTargetInitializer(values, p.getDefinition().getType());
+    initializer = ctypes.getTargetInitializer(values, p.getDefinition().getType());
+
+    // If Initializer is a code expression
+    // parse for parent parameter values being used inside expressions
+    // Look for <parent-param> encapsulated inside ${ <parent-param> }
```

Comment:
This `S{}` syntax for meta variables is something that is very specific to your tooling, so I don't think that this kind of substitution really belongs here. What I would do is find a way to bring the parameters into scope, so that the expressions are valid, and use the `${}` syntax in the initializers of the parameters themselves (not the expressions that reference them). Have you tried this?

Reply:
We're good as far as parent reactor's parameters are in the scope to have valid expressions, whichever syntax you propose. This ${} syntax is just to differentiate between child's bank_index and parent's bank_index. Syntax is your call, this change is not tied up with our tooling, this is purely lingua franca related.